### ISSUE TITLE
Returning an integer at a controller method causes server crashing

### ISSUE THREAD

**Issue by m0uneer** [Romakita/ts-express-decorators#80]
Hi Romain,

I'm facing problem returning integers from controller methods as the server process keep crashing.

**Current Version:** 1.4.12
**Log Stack:**

```yaml
[DEBUG] [#1] -- Incoming request --------------------------------------------------------
  [DEBUG] [#1] Route => DELETE /api/test/1
  [DEBUG] [#1] Request.id => 1
  [DEBUG] [#1] Headers => {"host":"192.168.1.101:3020","connection":"keep-alive","postman-token":"fc1e4ed3-bfba-ac63-b135-535b7bc04ce1","cache-control":"no-cache","origin":"chrome-extension://aicmkgpgakddgnaphhhpliifpcfhicfo","x-postman-interceptor-id":"be22a4bb-92b4-65e5-e183-9d9d88ea5cc6","user-agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.110 Safari/537.36","content-type":"application/json","accept":"*/*","accept-encoding":"gzip, deflate, sdch, br","accept-language":"en-US,en;q=0.8,ar;q=0.6,ru;q=0.4,nl;q=0.2","cookie":"connect.sid=s%3AOosP-oq2-XQ3gR6xsQeuG8YNrWc_Z1Kw.qZ98BjcW85Y1bnzs9e0tqeTRqCax%2FtIuQTVSfS8pWco"}
[DEBUG] [#1] ----------------------------------------------------------------------------
  [DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":false}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"cookieParser","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"cookieParser","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"compression","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"compression","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"jsonParser","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"jsonParser","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"urlencodedParser","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"urlencodedParser","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"session","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"session","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"bound initialize","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"bound initialize","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"bound authenticate","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"bound authenticate","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true,"returnPromise":true}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true}
[WARN ] [#1] [INVOKE][DUPLICATE] The handler have been resolved twice. See your code and find if you use @Next() and Promise at the same time.
  [WARN ] [#1] [INVOKE][DUPLICATE] Trace: {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true,"returnPromise":true}
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true,"returnPromise":true}
[DEBUG] [#1] Endpoint => {"target":"Ctrl","methodClass":"remove","httpMethod":"delete"}
[DEBUG] [#1] [INVOKE][START] {"type":"ENDPOINT","target":"Ctrl","methodName":"remove","injectable":true,"hasNextFn":false}
[WARN ] [#1] [INVOKE][DUPLICATE] The handler have been resolved twice. See your code and find if you use @Next() and Promise at the same time.
  [WARN ] [#1] [INVOKE][DUPLICATE] Trace: {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":true,"returnPromise":true}
  [DEBUG] [#1] [INVOKE][END  ] {"type":"ENDPOINT","target":"Ctrl","methodName":"remove","injectable":true,"hasNextFn":false,"returnPromise":true,"data":1}
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"SendResponseMiddleware","methodName":"use","injectable":true,"hasNextFn":false}
Thu, 27 Jul 2017 14:37:21 GMT express deprecated res.send(status): Use res.sendStatus(status) instead at node_modules/ts-express-decorators/lib/middlewares/send-response.js:33:22
  [DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"SendResponseMiddleware","methodName":"use","injectable":true,"hasNextFn":false}
error: uncaughtException: Invalid status code: 1 date=Thu Jul 27 2017 16:37:21 GMT+0200 (EET), pid=28490, uid=1000, gid=1000, cwd=server, execPath=.nvm/versions/node/v6.3.0/bin/node, version=v6.3.0, argv=[.nvm/versions/node/v6.3.0/bin/node, server/dist/app.js], rss=104169472, heapTotal=84955136, heapUsed=52815576, loadavg=[1.6103515625, 1.57763671875, 1.2587890625], uptime=360731
RangeError: Invalid status code: 1
at ServerResponse.writeHead (_http_server.js:192:11)
at ServerResponse.writeHead (server/node_modules/on-headers/index.js:55:19)
at ServerResponse.writeHead (server/node_modules/on-headers/index.js:55:19)
at ServerResponse._implicitHeader (_http_server.js:157:8)
at ServerResponse.end (server/node_modules/compression/index.js:102:14)
at writeend (server/node_modules/express-session/index.js:261:22)
at Immediate.onsave (server/node_modules/express-session/index.js:335:11)
at runCallback (timers.js:566:20)
at tryOnImmediate (timers.js:546:5)
at processImmediate [as _immediateCallback] (timers.js:525:5)
[DEBUG] [#1] [INVOKE][START] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":false}
[DEBUG] [#1] DELETE /api/test/1 undefined 485 ms
  [DEBUG] [#1] -- End Incoming request -----------------------------------------------------
  error: uncaughtException: Invalid status code: 1 date=Thu Jul 27 2017 16:37:21 GMT+0200 (EET), pid=28490, uid=1000, gid=1000, cwd=server, execPath=.nvm/versions/node/v6.3.0/bin/node, version=v6.3.0, argv=[.nvm/versions/node/v6.3.0/bin/node, server/dist/app.js], rss=104435712, heapTotal=84955136, heapUsed=53136424, loadavg=[1.6103515625, 1.57763671875, 1.2587890625], uptime=360731
RangeError: Invalid status code: 1
at ServerResponse.writeHead (_http_server.js:192:11)
at ServerResponse.writeHead (server/node_modules/on-headers/index.js:55:19)
at ServerResponse.writeHead (server/node_modules/on-headers/index.js:55:19)
at ServerResponse._implicitHeader (_http_server.js:157:8)
at ServerResponse.end (server/node_modules/compression/index.js:102:14)
at writeend (server/node_modules/express-session/index.js:261:22)
at Immediate.onsave (server/node_modules/express-session/index.js:335:11)
at runCallback (timers.js:566:20)
at tryOnImmediate (timers.js:546:5)
at processImmediate [as _immediateCallback] (timers.js:525:5)
[DEBUG] [#1] [INVOKE][END  ] {"type":"MIDDLEWARE","target":"anonymous","injectable":false,"hasNextFn":false}
```

Thanks for your continuous support.

Mouneer

**Comment by Romakita** [comment]
Hi @m0uneer,

Thanks for your issue. Effectively, when your return a number, Express thinks this data is the status parameters.  I never tried to return an number as response in a Rest API with Express. My bad :).

To fix quickly that, you can cast your response to a string. I'll push a new fix for that ASAP.

See you,
Romain

**Comment by Romakita** [comment]
fixed 1.4.13 :)

**Comment by m0uneer** [comment]
Great! Thanks Romain.
